openedx/core/djangoapps/enterprise/models.py

```diff
@@ -0,0 +1,175 @@
+        logo (ImageField):  Enterprise Customer logo
+        data_sharing_policy (CharField): stores data sharing policy: either no data sharing,
+                                         requested data sharing (learners are allowed to opt-in) or required data
+                                         sharing (learners must share their data)
+        sso_providers (CharField): list of comma-separated SSO provider slugs. Providers in this list should link
+                                    users registered with that provider with Enterprise Customer
+        active (BooleanField): used to mark inactive Enterprise Customers - implements "soft delete" pattern
+        members (ManyToMany[User]): list of Enterprise Customer members.
+    """
+    NO_DATA_SHARING = "no_sharing"
+    REQUEST_DATA_SHARING = "request"
+    REQUIRE_DATA_SHARING = "require"
+
+    class Meta(object):
+        """
+        Meta class for EnterpriseCustomer model
+        """
+        app_label = "enterprise"
+        verbose_name = "Enterprise Customer"
+
+    objects = EnterpriseCustomerManager()
+    all_objects = models.Manager()
+
+    name = models.CharField(max_length=500, blank=False, null=False, help_text=_(u"Enterprise Customer name."))
+    logo = models.ImageField(
+        upload_to="enterprise_logos",
+        help_text=_(u"Please add only .PNG files for logo images."),
+        null=True, blank=True, max_length=255
+    )
+    data_sharing_policy = models.CharField(
+        blank=False, null=False, help_text=_(u"Data sharing policy"), max_length=20,
+        choices=(
+            (NO_DATA_SHARING, _(u"No data sharing requested")),
+            (REQUEST_DATA_SHARING, _(u"Request data sharing")),
+            (REQUIRE_DATA_SHARING, _(u"Require data sharing")),
+        )
+    )
+    sso_providers = models.CharField(
+        blank=True, null=False, default="", max_length=500,
+        help_text=_(u"Comma-separated list of Single Sign On providers slugs.")
+    )
+    active = models.BooleanField(default=True)
+    # at a glance it seemed to be beneficial to use EnterpriseCustomerManager as related field manager here, so that
+    # only active enterprise customers are shown on the User side of relationship. However, Django docs explicitly
+    # mentions that it must not be done - see comment on EnterpriseCustomerManager
+    members = models.ManyToManyField(
+        User, related_name="enterprise_customer_membership", blank=True,
+        help_text=_(u"Members of Enterprise Customer.")
+    )
+
+    @property
+    def sso_providers_list(self):
+        """
+        List of SSO provider slugs
+        """
+        return [provider_slug.strip() for provider_slug in self.sso_providers.split(",")]
+
+    def __unicode__(self):
+        return u"<EnterpriseCustomer: {name}>".format(name=self.name)
+
+    def __repr__(self):
+        return self.__unicode__()
+
+
+class EnterpriseCustomerHistory(TimeStampedModel):
```

Comment:
Nit: since this model is immutable, we probably don't need the `modified` field, we only need `created`.